packages/react-router/tests/fileRoute.test-d.tsx

```diff
@@ -0,0 +1,87 @@
+import { expectTypeOf, test } from 'vitest'
+import { createFileRoute, createRootRoute } from '../src'
+
+const rootRoute = createRootRoute()
+
+const indexRoute = createFileRoute('/')()
+
+const invoicesRoute = createFileRoute('/invoices')()
+
+const invoiceRoute = createFileRoute('/invoices/$invoiceId')()
+
+const postLayoutRoute = createFileRoute('/_postLayout')()
+
+const postsRoute = createFileRoute('/_postLayout/posts')()
+
+const postRoute = createFileRoute('/_postLayout/posts/$postId_')()
+
+const authRoute = createFileRoute('/(auth)')()
+
+const protectedRoute = createFileRoute('/(auth)/protected')()
+
+declare module '../src/fileRoute' {
+  interface FileRoutesByPath {
```

Comment:
I think we should refractor the types for file based to allow us to pass `FileRoutesByPath` like we do for `RouteTree` in `Link`

Reply:
which changes would this involve?

Reply:
We would need to add a generic like `TRouteTree` so it can easily be more mocked in the tests